sdk/src/main/java/ly/count/android/sdk/ModuleDeviceId.java

```diff
@@ -192,6 +192,31 @@ void changeDeviceIdWithMergeInternal(@NonNull String deviceId) {
         }
     }
 
+    private void setIDInternal(String newDeviceID) {
+        if (Utils.isNullOrEmpty(newDeviceID)) {
+            L.w("[ModuleDeviceId] setID, Empty id passed to setID method");
+            return;
+        }
+
+        if (deviceIdInstance.getCurrentId() != null && deviceIdInstance.getCurrentId().equals(newDeviceID)) {
```

Comment:
this check is already inside the changeDeviceId functions?

Reply:
withMerge does not have it. Also I didn't want to pass bad value to the internal. I wanted to omit ASAP the bad value